fdbserver/DDTxnProcessor.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * DDTxnProcessor.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FOUNDATIONDB_DDTXNPROCESSOR_H
+#define FOUNDATIONDB_DDTXNPROCESSOR_H
+
+#include "fdbserver/Knobs.h"
+#include "fdbserver/DataDistribution.actor.h"
+
+class IDDTxnProcessor {
```

Comment:
I agree with the abstraction, but I'm wondering if this should be named something else - based on what it is rather than what it does. Something like IDDRegistry

Reply:
I'm not sure what's the best practice here. But my personal feeling is `IDDRegistry` doesn't tell me what code I should put into the implementation, it's a bit too general so that it might end up with everything in it, then the construction of this class might become a complex thing.
The good thing with `IDDTxnProcessor` is that it tells you it is just for running transactions, which limits the responsibility of this interface.